### ISSUE TITLE
[Enhance] Toggle's link will be a <button> - HTML

### ISSUE THREAD

**Issue by RCopeland** [sparkdesignsystem/spark-design-system#2938]
**User Story:**
As Spark we want Toggle to use `<button>` to open and close so that we are accessible. 

**Notes:**
Current State: Toggle uses a link to open and close it, since it doesn't navigate, it should be a button. 
Requestor: A11y audit, Section RM-12.
SME: Bob

**AC:**
Vanilla Toggle
1. will see `<a>` replaced to use `<button>` and fix the CSS.
2. will have aria-controls. (Bob Note: We can bake this into toggle.js and include a lodash call if the content item doesn't already have an id on it.)

`aria-controls` this attribute is on the trigger element (the button). It's value is set to the id of the content element. If the user provides an id use that. If the user doesn't provide an id, generate one with lodash.
